### ISSUE TITLE
Something always fails because of codemeta

### ISSUE THREAD

**Issue by broeder-j** [Materials-Data-Science-and-Informatics/somesy#50]
**Describe the bug**
In my case somesy always fails in codemeta, bevause `packagename = pkg.name`

happens in all cases for me, through hooks, cli, etc.

What I found so far is:
https://stackoverflow.com/questions/67212594/attributeerror-pathdistribution-object-has-no-attribute-name
that it might have to do with the version of  importlib-metadata<3.4.0.

**To Reproduce**
1. Have a codemeta.json file
2. install somesy (pypi)
3. Run it

**Expected behavior**

**Traces**
Stack trace of the error
```
ERROR    Error: 'PathDistribution' object has no attribute 'name'                                                                                                                util.py:18
DEBUG    Traceback (most recent call last):                                                                                                                                      util.py:19
           File "/home/j.broeder/.cache/pypoetry/virtualenvs/data-harvesting-m9-r40y6-py3.9/lib/python3.9/site-packages/somesy/cli/util.py", line 15, in wrap_exceptions                   
             return wrapped(*args, **kwargs)                                                                                                                                               
           File "/home/j.broeder/.cache/pypoetry/virtualenvs/data-harvesting-m9-r40y6-py3.9/lib/python3.9/site-packages/somesy/cli/sync.py", line 146, in sync                             
             run_sync(somesy_input)                                                                                                                                                        
           File "/home/j.broeder/.cache/pypoetry/virtualenvs/data-harvesting-m9-r40y6-py3.9/lib/python3.9/site-packages/somesy/cli/sync.py", line 169, in run_sync                         
             sync_command(somesy_input)                                                                                                                                                    
           File "/home/j.broeder/.cache/pypoetry/virtualenvs/data-harvesting-m9-r40y6-py3.9/lib/python3.9/site-packages/somesy/commands/sync.py", line 39, in sync                         
             _sync_codemeta(conf)                                                                                                                                                          
           File "/home/j.broeder/.cache/pypoetry/virtualenvs/data-harvesting-m9-r40y6-py3.9/lib/python3.9/site-packages/somesy/commands/sync.py", line 98, in _sync_codemeta               
             if update_codemeta(conf):                                                                                                                                                     
           File "/home/j.broeder/.cache/pypoetry/virtualenvs/data-harvesting-m9-r40y6-py3.9/lib/python3.9/site-packages/somesy/codemeta/__init__.py", line 77, in                          
         update_codemeta                                                                                                                                                                   
             cm_harvest = gen_codemeta(cm_sources)                                                                                                                                         
           File "/home/j.broeder/.cache/pypoetry/virtualenvs/data-harvesting-m9-r40y6-py3.9/lib/python3.9/site-packages/somesy/codemeta/exec.py", line 25, in gen_codemeta                 
             g, res, args, _ = build(                                                                                                                                                      
           File "/home/j.broeder/.cache/pypoetry/virtualenvs/data-harvesting-m9-r40y6-py3.9/lib/python3.9/site-packages/codemeta/codemeta.py", line 656, in build                          
             codemeta.parsers.python.parse_python(newgraph, res, source, crosswalk, args)                                                                                                  
           File "/home/j.broeder/.cache/pypoetry/virtualenvs/data-harvesting-m9-r40y6-py3.9/lib/python3.9/site-packages/codemeta/parsers/python.py", line 226, in parse_python             
             packagename = pkg.name                                                                                                                                                        
         AttributeError: 'PathDistribution' object has no attribute 'name'                                                                                                                 ```                                                                           

** Env **

CodeMetaPy==2.5.1
importlib-metadata==3.4.0
python 3.9
somesy==0.1.0

**Comment by apirogov** [comment]
Do you get the same problem when you use `codemetapy` directly?

`somesy` is only wrapping codemetapy and this error looks like it comes from inside the codemetapy parsing pipeline.

**Comment by apirogov** [comment]
The problem you have is unrelated to `somesy`, you get the same error when running `codemetapy` directly on your [pyproject.toml](https://codebase.helmholtz.cloud/hmc/hmc-public/unhide/data_harvesting/-/blob/main/pyproject.toml).

```
No input files specified, but found python project (pyproject.toml) in current dir, using that...
Note: You did not specify a --baseuri so we will not provide identifiers (IRIs) for your SoftwareSourceCode resources (and others)
Initial URI automatically generated, may be overriden later: file:///pyproject-toml
Processing source #1 of 1
Obtaining python package metadata for: pyproject.toml
Loading metadata from pyproject.toml via pyproject-parser
Failed to process pyproject.toml via pyproject-parser: The 'project.license' table should contain one of 'text' or 'file'.
Fallback: Loading metadata from pyproject.toml via PEP517

[notice] A new release of pip is available: 23.2.1 -> 23.3.1
[notice] To update, run: pip install --upgrade pip
Traceback (most recent call last):
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/testcmpy-m3kZsOkJ-py3.9/bin/codemetapy", line 8, in <module>
    sys.exit(main())
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/testcmpy-m3kZsOkJ-py3.9/lib/python3.9/site-packages/codemeta/codemeta.py", line 335, in main
    g, res, args, contextgraph = build(**args.__dict__)
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/testcmpy-m3kZsOkJ-py3.9/lib/python3.9/site-packages/codemeta/codemeta.py", line 656, in build
    codemeta.parsers.python.parse_python(newgraph, res, source, crosswalk, args)
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/testcmpy-m3kZsOkJ-py3.9/lib/python3.9/site-packages/codemeta/parsers/python.py", line 226, in parse_python
    packagename = pkg.name
AttributeError: 'PathDistribution' object has no attribute 'name'
```

There you can see that your `project.license` is wrong for the setuptools format, it must be `license = {text = "MIT"}`.

Fixing that gives a different error message:

```
No input files specified, but found python project (pyproject.toml) in current dir, using that...
Note: You did not specify a --baseuri so we will not provide identifiers (IRIs) for your SoftwareSourceCode resources (and others)
Initial URI automatically generated, may be overriden later: file:///pyproject-toml
Processing source #1 of 1
Obtaining python package metadata for: pyproject.toml
Loading metadata from pyproject.toml via pyproject-parser
Traceback (most recent call last):
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/testcmpy-m3kZsOkJ-py3.9/bin/codemetapy", line 8, in <module>
    sys.exit(main())
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/testcmpy-m3kZsOkJ-py3.9/lib/python3.9/site-packages/codemeta/codemeta.py", line 335, in main
    g, res, args, contextgraph = build(**args.__dict__)
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/testcmpy-m3kZsOkJ-py3.9/lib/python3.9/site-packages/codemeta/codemeta.py", line 656, in build
    codemeta.parsers.python.parse_python(newgraph, res, source, crosswalk, args)
  File "/local/home/a.pirogov/.cache/pypoetry/virtualenvs/testcmpy-m3kZsOkJ-py3.9/lib/python3.9/site-packages/codemeta/parsers/python.py", line 339, in parse_python
    if value.startswith("http"):
AttributeError: 'Readme' object has no attribute 'startswith'
```

I also want to note that it is strange that you have both "setuptools" and "poetry" style metadata. This definitely would cause issues with somesy and I can imagine also confuse other tools.

But in any case, this is unrelated to somesy, so please open an issue for codemetapy should you confirm that there is in fact a bug and your `pyproject.toml` file is valid.

**Comment by broeder-j** [comment]
Thanks. 

I know. I would only want the project part, not the poetry part, but since poetry fails to use the project part, I need to duplicate it, that poetry is aware of it. Other tools usually read and should only read the usual project part in my opinion. One reason for me not to use poetry.

The license thing we have corrected badly when enabling somesy, since before it was dynamic.
